Domains/4-Application/Connectors/GeologicalModel/GeologicalModel.ecschema.xml

```diff
@@ -0,0 +1,46 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<ECSchema schemaName="GeologicalModel" alias="geomodel" version="01.00.00" xmlns="http://www.bentley.com/schemas/Bentley.ECXML.3.2" displayLabel="Geological Model" description="The GeologicalModel schema defines classes that represent data for interpreting the earth's subterranean structure">
+    <ECSchemaReference name="BisCore" version="01.00.14" alias="bis" />
+    <ECSchemaReference name="CoreCustomAttributes" version="01.00.03" alias="CoreCA"/>
+    <ECSchemaReference name="Analytical" version="01.00.00" alias="anlyt" />
+
+    <ECEntityClass typeName="GeologicAnalyticalPartition" modifier="Sealed" description="A specialized Analytical Partition advertising a Geologic model (interpretation) based on Fault Blocks partitioned by one or more Fault Surfaces.">
+        <BaseClass>anlyt:AnalyticalPartition</BaseClass>
+    </ECEntityClass>
+
+    <ECEntityClass typeName="GeologicalModel" modifier="Sealed" description="An AnalyticalModel specialization expected to contain GeologicElement3d instances at different granularities.">
+        <BaseClass>anlyt:AnalyticalModel</BaseClass>
+    </ECEntityClass>
+
+    <ECEntityClass typeName="GeologicElement3d" modifier="Abstract" description="An AnalyticalElement serving as common base class for all element-classes modeling Geological concepts.">
+        <BaseClass>anlyt:AnalyticalElement</BaseClass>
+    </ECEntityClass>
+
+    <ECEntityClass typeName="FaultBlock" modifier="None" description="A GeologicElement3d modeling a set of volumes and surfaces that may be partitioned by one or more Fault Surfaces">
+        <BaseClass>GeologicAnalyticalPartition</BaseClass>
```

Comment:
@diegoalexdiaz the `FaultBlock` itself does not have any geometry associated. It's simply a grouping of `GeologicElement3D` instances. Would it make sense to inherit from the `GeologicAnalyticalPartition` instead?

Reply:
@danywu no, it wouldn't make sense. The purpose of InformationPartition classes in BIS is to advertise the modeling-perspective that a branch below a Subject focuses on. The GeologicAnalyticalPartition class is doing just that, so it is left 'sealed'. For more info on the purpose of InformationPartitions, see https://ddiazbisdocs.z13.web.core.windows.net/bis/guide/data-organization/modeling-perspectives/

On the other hand, a FaultBlock is modeling a specific concept in your domain. Thus, it needs to be a subclass of AnalyticalElement (via GeologicElement3d). It doesn't matter if such element typically doesn't have geometry on its own. It is fine that their `GeometryStream` property is left NULL.

I think the "Partition" wording of the InformationPartition classes in BIS may be introducing confusion in your domain as such word has a strong meaning in your world. Note that InformationPartition classes are typically not shown in the UX so they won't generate confusion to end-users. Their purpose is rather aimed at software-discoverable semantics (e.g. a piece of software can traverse the Subject hierarchy and find the InformationPartitions that it is designed to handle - e.g. Physical or Documents or Definitions or a specific kind of Analysis).